common/src/main/java/com/wynntils/features/user/ContainerSearchFeature.java

```diff
@@ -121,34 +121,41 @@ private void tryAutoSearch(AbstractContainerScreen<?> abstractContainerScreen) {
         String name = ComponentUtils.getCoded(abstractContainerScreen
                 .getMenu()
                 .getItems()
-                .get(currentContainerType.getNextItemSlot())
+                .get(currentSearchableContainerType.getNextItemSlot())
                 .getHoverName());
 
-        if (!currentContainerType.nextItemPattern.matcher(name).matches()) {
+        if (!currentSearchableContainerType.getNextItemPattern().matcher(name).matches()) {
             autoSearching = false;
             return;
         }
 
         ContainerUtils.clickOnSlot(
-                currentContainerType.getNextItemSlot(),
+                currentSearchableContainerType.getNextItemSlot(),
                 abstractContainerScreen.getMenu().containerId,
                 GLFW.GLFW_MOUSE_BUTTON_LEFT,
                 abstractContainerScreen.getMenu().getItems());
     }
 
-    private ContainerType getCurrentContainerType(String title) {
-        if (filterInBank && ContainerType.BANK.getTitlePattern().matcher(title).matches()) {
-            return ContainerType.BANK;
+    private SearchableContainerType getCurrentSearchableContainerType(String title) {
```

Comment:
Can't you move this as well to the SearchableContainerType class, and just keep the `filterInBank` etc checks here?

Reply:
Also, that means you don't have to expose the pattern from the ContainerType class